src/core/include/units/unit.h

```diff
@@ -462,9 +478,232 @@ inline constexpr decltype(U * U) square;
 template<Unit auto U>
 inline constexpr decltype(U * U * U) cubic;
 
+
+// get_unit_symbol
+
+enum class text_encoding {
+  unicode,  // m³;  µs
+  ascii,    // m^3; us
+  default_encoding = unicode
+};
+
+enum class unit_symbol_denominator {
+  solidus_one,      // m/s;   kg m-1 s-1
+  always_solidus,   // m/s;   kg/(m s)
+  always_negative,  // m s-1; kg m-1 s-1
+  default_denominator = solidus_one
+};
+
+enum class unit_symbol_separator {
+  space,  // kg m²/s²
+  dot,    // kg⋅m²/s²  (valid only for unicode encoding)
+  default_separator = space
+};
+
+struct unit_symbol_formatting {
+  text_encoding encoding = text_encoding::default_encoding;
+  unit_symbol_denominator denominator = unit_symbol_denominator::default_denominator;
+  unit_symbol_separator separator = unit_symbol_separator::default_separator;
+};
+
+namespace detail {
+
+// TODO Should `basic_symbol_text` be fixed to use `char` type for both encodings?
+template<typename CharT, typename UnicodeCharT, std::size_t N, std::size_t M, std::output_iterator<CharT> Out>
+constexpr Out copy(const basic_symbol_text<UnicodeCharT, N, M>& txt, text_encoding encoding, Out out)
+{
+  if (encoding == text_encoding::unicode) {
+    if (is_same_v<CharT, UnicodeCharT>)
+      return copy(txt.unicode(), out).out;
+    else
+      static_assert("Unicode text can't be copied to CharT output");
+  } else {
+    if (is_same_v<CharT, char>)
+      return copy(txt.ascii(), out).out;
+    else
+      static_assert("ASCII text can't be copied to CharT output");
+  }
```

Comment:
These assertions are never `false`.

Reply:
Thanks! It seems I am getting tired ;-)
Fixed!